commit 0fea615e526b4b7eff0363ee02d5753e5f924089
Author: Harald Welte <HaraldWelte@viatech.com>
Date:   Mon Jun 8 18:29:36 2009 +0800

    CPUFREQ: Mark e_powersaver driver as EXPERIMENTAL and DANGEROUS
    
    The e_powersaver driver for VIA's C7 CPU's needs to be marked as
    DANGEROUS as it configures the CPU to power states that are out
    of specification.
    
    According to Centaur, all systems with C7 and Nano CPU's support
    the ACPI p-state method.  Thus, the acpi-cpufreq driver should
    be used instead.
    
    Signed-off-by: Harald Welte <HaraldWelte@viatech.com>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

diff --git a/arch/x86/kernel/cpu/cpufreq/Kconfig b/arch/x86/kernel/cpu/cpufreq/Kconfig
index 52c8398..f138c6c 100644
--- a/arch/x86/kernel/cpu/cpufreq/Kconfig
+++ b/arch/x86/kernel/cpu/cpufreq/Kconfig
@@ -220,11 +220,14 @@ config X86_LONGHAUL
 	  If in doubt, say N.
 
 config X86_E_POWERSAVER
-	tristate "VIA C7 Enhanced PowerSaver"
+	tristate "VIA C7 Enhanced PowerSaver (DANGEROUS)"
 	select CPU_FREQ_TABLE
-	depends on X86_32
+	depends on X86_32 && EXPERIMENTAL
 	help
-	  This adds the CPUFreq driver for VIA C7 processors.
+	  This adds the CPUFreq driver for VIA C7 processors.  However, this driver
+	  does not have any safeguards to prevent operating the CPU out of spec
+	  and is thus considered dangerous.  Please use the regular ACPI cpufreq
+	  driver, enabled by CONFIG_X86_ACPI_CPUFREQ.
 
 	  If in doubt, say N.
 
